pkg/ruler/api.go

```diff
@@ -168,20 +177,20 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 		grp := RuleGroup{
 			Name:           g.Group.Name,
 			File:           g.Group.Namespace,
-			Rules:          make([]rule, len(g.ActiveRules)),
+			Rules:          []rule{},
 			Interval:       g.Group.Interval.Seconds(),
 			LastEvaluation: g.GetEvaluationTimestamp(),
 			EvaluationTime: g.GetEvaluationDuration().Seconds(),
 			SourceTenants:  g.Group.GetSourceTenants(),
 		}
 
 		for i, rl := range g.ActiveRules {
-			if g.ActiveRules[i].Rule.Alert != "" {
+			if g.ActiveRules[i].Rule.Alert != "" && returnAlerts {
```

Comment:
Perhaps this logic would be clearer with `g.ActiveRules[i].Rule.Alert != ""` extracted to a `isAlertRule` variable? (and similarly for recording rules below)

Reply:
This is no longer relevant since the condition was moved to the rulers IMHO